ddclient.in

```diff
@@ -2146,6 +2154,128 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## geturl -- via curl
+######################################################################
+sub curl_cmd {
+    my $params = join("\n",@_);
+    my $tmpfile;
+
+    return '' if (length($params) == 0); ## no parameters provided
+
+    # debug("CURL:\r\n%s\r\n",$params); ## Uncomment only if required as may expose credentials
+
+    eval "no warnings 'all'; use File::Spec;";
+    $tmpfile = ((!$@) ? File::Spec->tmpdir() : '/tmp').'/ddclient-';
+    my @chars = ("A".."Z", "a".."z", "0".."9");
+    $tmpfile .= $chars[rand @chars] for 1..10;
+    debug("CURL Tempfile: %s",$tmpfile);
+    open(my $TFH, '>', $tmpfile) or fatal("Falied to open $tmpfile for curl");
```

Comment:
Use [File::Temp](https://perldoc.perl.org/File/Temp.html).

Better would be to use standard input via something like IPC::Open2.

Reply:
I had an implementation that preferred File::Temp and used it only if it existed.  The problem is that it auto deletes the file when you're done with it so when I do the unlink I get warning message that the file is gone already.  But in the case where File::Temp does not exist on the system (rare) you must do the unlink.  I tried everything to suppress that warning but could not get rid of it.  But I can put it back if you want.

IPC::Open2 installed by default on even fewer systems that File::Temp.

Reply:
File::Temp is a core module and exists in Perl v5.10.1, so it is safe to assume it exists.

Reply:
How would you like it to fail (or not fail) if File::Temp doesn't exist.
@abelbeck FYI as affects astlinux which does not have this -- would you add it?

Reply:
Add it to configure.ac and let them handle it.

Reply:
I have added File::Temp.  I'm not sure about handling in configure... we have to cater for the cross-compile use case (building a system on one device for a different target).  This would require something like `--with[out]-file-temp` support which would then have to set some values which would then have to change the code between `ddclient.in` and `ddclient` which I think is bad (as it is, I still don't understand why we need a `.in` version and can't just use `ddclient`). @abelbeck FYI.

Reply:
>  I'm not sure about handling in configure

Just add it to the [list of required Perl modules](https://github.com/ddclient/ddclient/blob/ca1b519444aafe346341800ad80765b72aca0efb/configure.ac#L40-L42) under "Perl modules required to run ddclient."

> we have to cater for the cross-compile use case

`configure.ac` is not set up for that—it assumes that you are running `configure` on the target system itself. If you need cross-build support, please open a new issue. In the meantime, you can patch `configure.ac` to delete the module checks.

> I still don't understand why we need a `.in` version and can't just use `ddclient`

Reason 1: We need a way to substitute user-supplied values that can differ between systems for technical or policy reasons. Examples include:
  * The filename of the `perl` executable.
  * Which directory contains `ddclient.conf`.
  * Where the cache file should be written.

Reason 2: We want to move the documentation source out of `ddclient.in` and into a separate XML or Markdown file so that we can easily build a man page. The documentation will be converted to plain text and substituted back into the resulting `ddclient` script.

Reason 3: In the future we might want to use something like [`pp`](https://metacpan.org/pod/pp) to package the script and its required modules into a single executable.

Reply:
Added File::Temp to list of required modules.